clients/src/main/resources/common/message/ListGroupsRequest.json

```diff
@@ -23,11 +23,15 @@
   // Version 3 is the first flexible version.
   //
   // Version 4 adds the StatesFilter field (KIP-518).
-  "validVersions": "0-4",
+  //
+  // Version 5 adds the TypesFilter field (KIP-848).
+  "validVersions": "0-5",
   "flexibleVersions": "3+",
   "fields": [
     { "name": "StatesFilter", "type": "[]string", "versions": "4+",
       "about": "The states of the groups we want to list. If empty all groups are returned with their state."
-    }
+    },
+    { "name": "TypesFilter", "type": "[]string", "versions": "5+",
+      "about": "The types of the groups we want to list. If empty all groups are returned" }
```

Comment:
I would rather prefer to do this in a second PR because this change impacts both the new and the old group coordinators. Would it be possible?

Reply:
makes sense

Reply:
TODO: Add a second pr to update ListGroups protocol

Reply:
The test will use the code generated by the new protocol. If the protocol change is to be placed in the second pr, the corresponding test code also needs to be removed in this pr.
I don't know if my understanding is correct

Reply:
If the tests are related to the new filter, they should go to the next PR.